xbmc/addons/Addon.cpp

```diff
@@ -264,6 +264,59 @@ bool CAddon::SettingsLoaded(AddonInstanceId id /* = ADDON_SETTINGS_ID */) const
   return addonSettings && addonSettings->IsLoaded();
 }
 
+class FilenameXMLCache 
+{
+public:
+	const CXBMCTinyXML* loadXMLFile(const std::string& id, const std::string& xmlFilename)
+	{
+	  struct __stat64 s;
+	  if (XFILE::CFile::Stat(xmlFilename, &s) == 0)
+	  {
+		auto found = cache.find(xmlFilename);
+		if (found != cache.end() && s.st_mtime <= found->second.modified)
+		{
+		  auto& cachedItem = found->second;
+		  return &cachedItem.xml;
+		}
+		else
+		{
+		  auto& cachedItem = cache[xmlFilename];
+		  cachedItem.modified = s.st_mtime;
+
+		  if (!cachedItem.xml.LoadFile(xmlFilename))
+		  {
+			if (CFile::Exists(xmlFilename))
+			{
+			  CLog::Log(LOGERROR, "CAddon[{}]: unable to load: {}, Line {}\n{}", id, xmlFilename,
+						cachedItem.xml.ErrorRow(), cachedItem.xml.ErrorDesc());
+			}
+			cache.erase(xmlFilename);
+			return nullptr;
+		  }
+
+		  return &cachedItem.xml;
+		}
+	  }
+	  return nullptr;
+	}
+
+private:
+
+	struct CacheItem
+	{
+	  time_t modified;
+	  CXBMCTinyXML xml;
+	};
+
+	std::unordered_map<std::string, CacheItem> cache;
+};
+
+const CXBMCTinyXML* loadXMLFile(const std::string& id, const std::string& xmlFilename)
+{
+	static FilenameXMLCache cache;
```

Comment:
It looks to me that you're globally caching xml files from different addons on the same place. What happens when a given addon (that has contributed to the cache) is uninstalled? I suspect you keep those objects cached regardless :)

Reply:
As this requires only a couple lines of code I suggest to do it even if maybe not strictly necessary. Suggestion fo an implementation: https://github.com/neo1973/xbmc/commit/0b062766319ea1689c4d9bdf66dca846597df380 (not runtime tested)